source/TypeTopology/AbsolutenessOfCompactness.lagda

```diff
@@ -0,0 +1,403 @@
+Andrew Swan, February 7th 2024
+
+This is a generalisation of some of the results by Martín Escardó in
+TypeTopology.PropTychonoff, based on the observation that for
+propositions P, the functor sending A to P → A is a
+modality. Modalities of this form are an important special case and
+they have a name; they are *open modalities* (Example 1.7 in
+[1]). However, we will show a version of the theorem is not only true
+for open modalities, but for all modalities.
+
+For another example, let ∇ be the modality of double negation sheaves
+(Example 3.41 of [1]). The internal logic in this reflective universe
+is boolean. It follows that ∇ (is-compact∙ A) holds for all types A,
+and so we can deduce that ∇ A is always compact.
+
+TODO: add double negation sheaves to the library
+
+We can also see as a special case that truncation preserves
+compactness, although it seems unlikely there are any good examples of
+compact higher types where it isn't already clear that the
+0-truncation is compact.
+
+When formulated in terms of modalities, the result is best thought of
+as an "absoluteness result." When working in models of some theory, a
+logical formula might make sense stated both internally in the model
+and for the same object viewed externally from outside the
+model. Absoluteness says that these two statements are equivalent.
+
+In this case we are thinking of reflective subuniverses as models of
+type theory sitting inside some larger "external" model of type
+theory. We will show that compactness is an upwards absolute
+notion. That is, if a type inside the reflective subuniverse is
+compact with respect to the internal logic of the subuniverse then it
+is compact viewed outside the subuniverse as just a type. The converse
+does not quite hold, so potentially there could be compact types where
+the internal statement of compactness is not true, and we don't get
+full absoluteness.
+
+We note that the result holds for all modalities with no further
+conditions and in particular the modality is not required to be lex,
+or to preserve 𝟘. For the main theorem, we don't even need a full
+modality, and the weaker notion of reflective subuniverse suffices.
+
+[1] Rijke, Shulman, Spitters, Modalities in homotopy type theory,
+https://doi.org/10.23638/LMCS-16(1:2)2020
+
+\begin{code}
+{-# OPTIONS --safe --without-K --exact-split --auto-inline #-}
```

Comment:
The options `exact-split` and `auto-inline` are in the lib file and so are not needed. So are the other two, but I prefer to keep them for emphasis in all files.

Reply:
So, in summary, keep only the first two options.